### DISCUSSION TITLE
Any way to force a new row in a grid?

### DISCUSSION THREAD

**Discussion by muteboy** [terrastruct/d2#1807] [Q&A] [answered]
Using a grid, I want to specify 2 rows. The first row is the header, the second row contains all the entries, like so:
![image](https://github.com/terrastruct/d2/assets/10484438/c15ab41c-7bd5-44a9-a1d3-12688c97f441)

But to do this I would need to specify when the second row would start. Can I achieve this using grids?

**Comment by alixander** [comment]
Hi @muteboy you can achieve this by fitting dimensions

https://play.d2lang.com/?script=NMtBCsIwEIXh_ZziUXcFJa1mkyN4iykZmoI2MhkJIt5dRGb3PXi_8XKThGHAm4BVt3xc-ZEQfGntLWEmYDz1LVtJmEIgAg64PpuhcINVLIJ7VYEV3n-PMRJQhLOod_HfyW76mhyz4-y4OCJ96BsAAP__&

```d2
table: "" {
  grid-gap: 0
  grid-rows: 2
  *.width: 100

  # Just has to be >= 100*5
  header.width: 500

  entry1
  entry2
  entry3
  entry4
  entry5
}
```
![d2 (41)](https://github.com/terrastruct/d2/assets/3120367/710ba87d-caf2-413c-b181-bfe65988e19d)

**Comment by muteboy** [comment]
Thanks for the quick reply!
I should have said that I don't know how many items there will be in the second row. I will be generating the D2 markup, with multiple tables, each with a header, and then 1 or more items in the second row.

But I've just realised there's no way (that I can see) to have more than one table in the diagram. I thought I could generate multiple tables and have them be separate, but that doesn't seem to be the case.

Not to worry, I can just use the SQL Table instead, that meets most of my needs.

**Reply by alixander** [reply]
> I will be generating the D2 markup, with multiple tables, each with a header, and then 1 or more items in the second row.

If you're generating the D2 markup programmatically in an environment where there's math, you can always just make the header row multiplied by elements.

> But I've just realised there's no way (that I can see) to have more than one table in the diagram.

I'm not sure what you mean. Why not?

https://play.d2lang.com/?script=zM5BCsJADIXhfU7xqLuC0o7OZo7gLVImdArakUykiHh3Uckd3H2BPPiNp4skdB2eBMy65P3Mt4TBL61bSwgE9IdtyVYSxmEgAnY435uhcINVTIJrVYEVXj8ffSSgCGdR38XfTlbTx-gIjqPj5Ij0IvoGhj8ufAcAAP__&

<img width="1393" alt="Screen Shot 2024-01-15 at 9 55 02 PM" src="https://github.com/terrastruct/d2/assets/3120367/2c0df791-070b-44ea-aa40-50f031de3ea4">

**Comment by muteboy** [comment]
Thanks again! I really need to read the docs more closely :)
I thought "table:" created a table object, but you've shown me that it is the name of the object - so each table needs a unique name.

**Comment by bo-ku-ra** [comment, marked as answer]
cf.
```d2
table: "" {
  grid-gap: 0
  grid-columns: 1
  th: "title"
  tr1: "" {
    grid-gap: 0
    grid-rows: 1
    x
    xx
    xxx
    xxxx
    xxxxx
  }
}
```

**Reply by muteboy** [reply]
So nest a grid inside another grid? That works well, thank you.
